core/src/main/java/com/linecorp/armeria/common/logging/RequestScopedMdc.java

```diff
@@ -219,7 +220,9 @@ public static void put(RequestContext ctx, String key, @Nullable String value) {
         requireNonNull(ctx, "ctx");
         requireNonNull(key, "key");
 
-        synchronized (ctx) {
+        final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
I guess we need to apply a similar CAS operation as done [here](https://github.com/line/armeria/pull/4578#discussion_r1060368803) to the rest of the "locks" in this class

Reply:
i fixed it all👍 [377aa66](https://github.com/line/armeria/pull/4578/commits/377aa66e2f5b3509d7f4a3489b722ef9deb57532)